File: src/main/java/com/qsoftware/bubbles/effect/AttackBoostEffect.java
package com.qsoftware.bubbles.effect;

import com.qsoftware.bubbles.common.AttributeMap;
import com.qsoftware.bubbles.common.effect.Effect;
import com.qsoftware.bubbles.common.effect.EffectInstance;
import com.qsoftware.bubbles.common.entity.Attribute;
import com.qsoftware.bubbles.common.entity.Entity;
import com.qsoftware.utilities.python.builtins.ValueError;

public class AttackBoostEffect extends Effect<AttackBoostEffect> {
    public AttackBoostEffect() throws ValueError {
        super();
    }

    @Override
    public void execute(Entity evt, EffectInstance effectInstance) {

    }

    @Override
    protected void updateStrength(EffectInstance effectInstance, int old, int _new) {

    }

    @Override
    protected boolean canExecute(EffectInstance effectInstance) {
        return false;
    }

    @Override
    public AttributeMap getAttributeModifiers() {
        AttributeMap attributes = new AttributeMap();
        attributes.set(Attribute.ATTACK, 1f);

        return attributes;
    }
}

File: src/main/kotlin/qtech/bubbles/effect/AttackBoostEffect.kt
package qtech.bubbles.effect

import qtech.bubbles.common.AttributeMap
import qtech.bubbles.common.effect.Effect
import qtech.bubbles.common.effect.EffectInstance
import qtech.bubbles.common.entity.Attribute
import qtech.bubbles.entity.Entity

class AttackBoostEffect : Effect() {
    override fun execute(entity: Entity, effectInstance: EffectInstance) {}
    override fun updateStrength() {}
    override fun canExecute(entity: Entity, effectInstance: EffectInstance): Boolean {
        return false
    }

    override val attributeModifiers: AttributeMap
        get() {
            val attributes = AttributeMap()
            attributes.set(Attribute.ATTACK, 1f)
            return attributes
        }
}